File: app/src/main/java/com/franjo/android/bakingapp/data/di/RepositoryModule.java
package com.franjo.android.bakingapp.data.di;

import com.franjo.android.bakingapp.data.network.RecipeApiService;
import com.franjo.android.bakingapp.data.repository.RecipeRepositoryDataSourceImpl;
import com.franjo.android.bakingapp.domain.repository.IRecipeRepository;
import com.franjo.android.bakingapp.utilities.mapper.IModelMapper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class RepositoryModule {

    @Provides
    @Singleton
    IRecipeRepository provideRepositoryImpl(
            RecipeApiService recipeApiService, IModelMapper modelMapper) {
        return new RecipeRepositoryDataSourceImpl(recipeApiService, modelMapper);
    }
}

File: app/src/main/java/com/franjo/android/bakingapp/data/di/RepositoryModule.kt
package com.franjo.android.bakingapp.data.di

import com.franjo.android.bakingapp.data.repository.RecipeRepositoryImpl
import com.franjo.android.bakingapp.domain.repository.RecipeRepository
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
interface RepositoryModule {

    @Binds
    @Singleton
    fun bindRecipeRepository(
        impl: RecipeRepositoryImpl
    ): RecipeRepository

}
